library/options.inc.php

```diff
@@ -122,7 +122,33 @@ function generate_select_list($tag_name, $list_id, $currvalue, $title, $empty_na
 	$selectEmptyName = xlt($empty_name);
 	if ($empty_name)
 		$s .= "<option value=''>" . $selectEmptyName . "</option>";
-	$lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ? AND activity=1 ORDER BY seq, title", array($list_id));
+
+        // List order depends on language translation options.
+        //  (Note we do not need to worry about the list order in the algorithm
+        //   after the below code block since that is where searches for exceptions
+        //   are done which include inactive items or items from a backup
+        //   list; note these will always be shown at the bottom of the list no matter the
+        //   chosen order.)
+        $lang_id = empty($_SESSION['language_choice']) ? '1' : $_SESSION['language_choice'];
+        if ($GLOBALS['gb_how_sort_list'] == '0') {
+          //sort by seq
+          $lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ? AND activity=1 ORDER BY seq, title", array($list_id));
+        } else if ($GLOBALS['gb_how_sort_list'] == '1') {
+          // sort by title
+          if (($lang_id == '1' && !empty($GLOBALS['skip_english_translation'])) || !$GLOBALS['translate_lists']) {
+            $lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ? AND activity=1 ORDER BY title, seq", array($list_id));
+          } else {
+            $lres = sqlStatement("SELECT lo.option_id, " .
+              "IF(LENGTH(ld.definition),ld.definition,lo.title) AS title, " .
```

Comment:
Hi brady.
We must add to this query = "SELECT lo.option_id**, lo.is_default**......" the our second fix of the default option doesn't work without this column..